public/app/core/services/backend_srv.ts

```diff
@@ -212,6 +213,14 @@ export class BackendSrv implements BackendService {
               const firstAttempt = i === 0 && options.retry === 0;
 
               if (error.status === 401 && isLocalUrl(options.url) && firstAttempt && isSignedIn) {
+                if (error.data.message.includes('revoked')) {
+                  appEvents.emit(CoreEvents.showModalReact, {
+                    component: TokenRevokedModal,
+                  });
+
+                  return of(null);
```

Comment:
Is that correct? @Clarity-89

If I understood it correctly, I should return an `Observable` that returns an error `throwError` in case I want to retry it. But in this case I don't want to retry it neither have something to return 🤔

Reply:
Hmm, not sure this is the best place or way to handle this, matching by a part of error message seems too brittle.  I think @torkelo and @hugohaggmark have way more experience with the `backend_srv` though :)

Reply:
I agree this is a weak way of determining this. Is there a good reason why we need to show the Modal?

Reply:
> I agree this is a weak way of determining this. Is there a good reason why we need to show the Modal?

Yeah, the modal is a UX requirement related with license enforcement, in order to bring some feedback when the session is revoked due to reaching the maximum amount of concurrent sessions limit. You can find more information at https://github.com/grafana/grafana-enterprise/issues/745.

I added it here because it seemed to be the piece of code responsible for handling those situations (we no longer want retries / loginPing when there's a revoked token error, but to display the modal instead, no matter what action / request is the user performing). 

But almost new with the frontend, so I'm completely open for feedback. How would you handle it?  

Thanks!!😇 🙏🏻

Reply:
Ok, thank you for that clarification. Could we try to look for a specific id in the message instead of searching for a specific string? Regarding what I would do, I left some comments on that earlier, but if you want I can push some changes to your branch?

Reply:
>  Could we try to look for a specific id in the message instead of searching for a specific string?

Yeah, searching for a specific string neither looks ideal for me, but AFAIK we don't have any error codes definition or something similar for this can of situations. Let me see if I can find a way to address it in a neater way. 

> Regarding what I would do, I left some comments on that earlier, but if you want I can push some changes to your branch?

Yup, I really appreciate your comments. I'm gonna work on that feedback. 
But, could you help me with the RxJS tests? It's my time first with that library so any advice or example is more than welcome 😇

Reply:
I've opened a PR here @joanlopez https://github.com/joanlopez/grafana/pull/27

Reply:
Awesome, thanks for your help!! 🙏🏻 😇